### PR TITLE
Add resource detectors to cart service

### PR DESCRIPTION
**Pull request by svrnm**
# Changes

Resource detection for cart service, right now it is only the container.id because it looks like no other resource detectors exist for .NET (yet)

Screenshot 2022-12-28 at 12 16 06

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [x] Appropriate documentation updates in the [docs](../docs/) folder

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -166,3 +166,4 @@ significant modifications will be credited to OpenTelemetry Authors.
 ([#648](https://github.com/open-telemetry/opentelemetry-demo/pull/648))
 * Add Jaeger-SPM-Config
 ([#655](https://github.com/open-telemetry/opentelemetry-demo/pull/655))
+* Add resource detectors to cart service
```
Comment (fatsheep9146) [inline CHANGELOG.md:169]: link of pr is missing

Comment (svrnm) [reply CHANGELOG.md:169]: fixed, TY.

**Review by fatsheep9146 — APPROVED** [review]

**Review by julianocosta89 — COMMENTED** [review]
Tried out locally and nothing was added to the Resource Attributes. I suspect this is not working on my Ubuntu, because a similar issue happened when testing #662

**Comment by julianocosta89** [comment]
Is it any specific version of docker required to this to work?
Ran `docker system prune -a`, re-built all the images, but still no success :/

**Review by julianocosta89 — APPROVED** [review]
Tested in my Windows and it worked.
Also using `docker`

**Comment by mviitane** [comment]
Doesn't work for me when using docker desktop on mac.

**Comment by svrnm** [comment]
This is probably due to a Limitation because of cgroup v2, there is a solution in the making for dotnet:

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/839

Java & NodeJS have that fixed already.

**Comment by cartersocha** [comment]
Thanks @svrnm. Lgtm and the dependency PR seems to be making progress. 

Merging.

**Pull request merged**